spanner/batch.go

```diff
@@ -136,7 +140,18 @@ func (t *BatchReadOnlyTransaction) PartitionReadUsingIndexWithOptions(ctx contex
 		Columns:          columns,
 		KeySet:           kset,
 		PartitionOptions: opt.toProto(),
-	})
+	}, gax.WithGRPCOptions(grpc.Header(&md)))
+
+	if GFELatencyMetricsEnabled && md != nil && t.txReadOnly.CommonTags != nil {
+		ctxGFE, errGFE := createContextForGFELatencyMetrics(ctx, t.txReadOnly)
+		if errGFE != nil {
+			return nil, errGFE
+		}
+		errGFE = captureGFELatencyStats(ctxGFE, md, "PartitionReadUsingIndexWithOptions")
+		if errGFE != nil {
+			return nil, errGFE
+		}
+	}
```

Comment:
This block of code (with some small variations) seem to be duplicated very many times in this PR. Would it be possible to extract that to a separate method with the RPC name as an argument and that returns an error (and possibly other arguments that I'm missing here now)?